lib/galaxy/authnz/custos_authnz.py

```diff
@@ -189,19 +189,21 @@ def callback(self, state_token, authz_code, trans, login_redirect_url):
         if custos_authnz_token is None:
             user = trans.user
             existing_user = trans.sa_session.query(User).filter_by(email=email).first()
+            # If there is only a single external authentication
+            # provider in use, trust the user provided and
+            # automatically associate.
+            # Equivalent to `fixed_delegated_auth` from `ConfigSerializer`
+            # TODO: Future work will expand on this and provide an
+            # interface for when there are multiple auth providers
+            # allowing explicit authenticated association.
+            fixed_delegated_auth = (
```

Comment:
Did you consider adding this to the core Galaxy `config` at load time?  This stuff is static after that point I believe and we could consolidate the logic there.